nautobot/core/views/mixins.py

```diff
@@ -1387,3 +1388,106 @@ def notes(self, request, *args, **kwargs):
             "active_tab": "notes",
         }
         return Response(data)
+
+
+class ComponentCreateViewMixin(NautobotViewSetMixin, mixins.CreateModelMixin):
+    create_form_class: Optional[type[Form]] = None  # TODO: required, declared Optional only to avoid a breaking change
+    form_class: Optional[type[Form]] = None  # TODO: required, declared Optional only to avoid a breaking change
+
+    def create(self, request, *args, **kwargs):
+        if request.method == "POST":
+            return self.perform_create(request, *args, **kwargs)
+
+        create_form = self.create_form_class(initial=request.GET)  # pylint: disable=not-callable
+        model_form = self.form_class(request.GET)  # pylint: disable=not-callable
+
+        return Response(
+            {
+                "template": self.create_template_name,
+                "component_type": self.queryset.model._meta.verbose_name,
+                "model_form": model_form,
+                "form": create_form,
+                "return_url": self.get_return_url(request),
+            },
+        )
+
+    # TODO: this conflicts with DRF's CreateModelMixin.perform_create(self, serializer) API
+    def perform_create(self, request, *args, **kwargs):  # pylint: disable=arguments-differ
```

Comment:
I think just giving this a different name, e.g. `_perform_create()` or similar would address the TODO here.